lib/netbuf/netbuf_ssl_internal.h

```diff
@@ -12,4 +12,25 @@ extern void * (* netbuf_write_ssl_func)(struct network_ssl_ctx *,
     const uint8_t *, size_t, size_t, int (*)(void *, ssize_t), void *);
 extern void (* netbuf_write_ssl_cancel_func)(void *);
 
+/**
+ * netbuf_read_init2(s, ssl):
+ * Create and return a buffered reader attached to socket ${s} which uses the
```

Comment:
If `${ssl}` is not NULL, the socket `${s}` is ignored.  (In fact I think it will always be `-1`...)